site/content/kapp/docs/develop/config.md

```diff
@@ -217,6 +222,16 @@ waitRules:
 
 `diffMaskRules` specify which field values should be masked in diff. By default `v1/Secret`'s `data` fields are masked. Currently only applied to `deploy` command.
 
+### preflightRules
+
+Available in v0.60.1+.
+
+`preflightRules` specify configuration for [preflight checks](preflight.md). Specifying the `name` of a preflight check enables it; additional configuration via the `config` field may be optionally provided. The contents of the `config` field are specific to each preflight check.
+
+The `--preflight` flag overrides the enabled setting in the configuration:
+* If a preflight check is omitted from the `--preflight` flag, it is disabled regardless of its presence in the configuration file.
```

Comment:
Based on [this](https://github.com/carvel-dev/kapp/pull/900/files#diff-424696d14808b68e202e535fc53ddfc20262790b23f3eb8d057a665ea69479ccR140) and the [test](https://github.com/carvel-dev/kapp/pull/900/files#diff-c55b40d43ffd8bf5ab782c17d750df0624f1f5c0558e03df4835ee16fa753971R260-R275), I was thinking that we can enable a preflight check from both the command line or the config (which I am fine with), but I am guessing that wasn't the intention?

Apologies for looking at the doc PR after merging the actual PR 😓

Reply:
That is the intention! The command line will override the configuration file. So, if `--preflight` is specified, only those mentioned in `--preflight` will be used, regardless of the config file.

Reply:
The idea is to have a default set of configuration and enablement for preflight checks. And then at run time, the user can choose which checks they want to run, if it's different than the default (i.e. those listed in the config).

Reply:
Understood, I think I missed the `someCheck: false` in one of the other test.